server/api/utils/singletons/k8s.py

```diff
@@ -600,21 +600,35 @@ def delete_secrets(
                 )
                 raise exc
 
-        secret_data = {}
+        if not k8s_secret.data:
+            logger.debug(
+                "No data found in the Kubernetes secret",
+                secret_name=secret_name,
+            )
+            return None
```

Comment:
if you return here you will not delete the k8s secret 🤔

Reply:
You’re right!